src/neptune/new/handler.py

```diff
@@ -172,8 +170,9 @@ def assign(self, value, wait: bool = False) -> None:
             if not attr:
                 self._container.define(self._path, value)
             else:
-                neptune_value = cast_value(value)
-                attr.process_assignment(neptune_value, wait)
+                if isinstance(value, Handler):
+                    value = ValueCopy(value)
```

Comment:
Why not `cast_value`?

Reply:
ok

Reply:
No, we just override value when `value` is of instance `Handler`, if not we use original one.